lib/state_machines/integrations/active_record.rb

```diff
@@ -565,7 +565,13 @@ def locale_path
 
         # Defines a new named scope with the given name
         def create_scope(name, scope)
-          lambda { |model, values| model.where(scope.call(values)) }
+          ->(model, values) { values.present? ? model.where(scope.call(values)) : model }
+        end
+
+        # Generates the results for the given scope based on one or more states to filter by
+        def run_scope(scope, machine, klass, states)
```

Comment:
this is the same method as the one from state_machines https://github.com/state-machines/state_machines/blob/309668998449ca6c348de809f34660d822bc626e/lib/state_machines/machine.rb#L2130-L2135 but I just add `compact`